### DISCUSSION TITLE
Getting Started

### DISCUSSION THREAD

**Discussion by manudev-1** [shinyorg/shiny#1349] [Q&A]
Hi guys, im trying to explore the word of BLE and Bluetooth,
and i want to realize a basic MAUI application which has just one button and when the user press the button the application scan the near Bluetooth devices and show them in a list in the application.

I think i have setted everything that should be setted (Permissions) and downloaded everything correctly.

So I find this amazing Library and take a look at the documentation:

![image](https://github.com/shinyorg/shiny/assets/83871565/4d43295a-53dc-4218-929a-6a9af17ce43d)

I copied this code on my application but as i was thinking it throw an exception:
`System.NullReferenceException: 'Object reference not set to an instance of an object.'`

How can i resolve this little problem?

**Comment by aritchie** [comment]
You can't just copy & paste it, the bleManager doesn't get set.  You have to inject or resolve it using dependency injection.  

You can use if you don't know how to use dependency injection.

```csharp
var ble = Shiny.Hosting.Host.GetService<IBleManager>();
```

**Reply by manudev-1** [reply]
Thank you for your answer, I tryed this code in the sequent way:

```
using Shiny.BluetoothLE;
using Shiny.Hosting;

namespace BLE_Dev
{
    public partial class MainPage : ContentPage
    {        
        public MainPage()
        {
            InitializeComponent();
        }

        private void btnScan_Clicked(object sender, EventArgs e)
        {
            var manager = Host.GetService<IBleManager>();
            var scanner = manager.Scan().Subscribe(scanResult =>
            {
                Label label = new Label();
                label.Text = scanResult.Peripheral.Name;
                Devices.Add(label);
            });
        }
    }
}
```
But when i press the button it still gives me the error:
`System.NullReferenceException: 'Object reference not set to an instance of an object.'`

did i miss something?

**Reply by aritchie** [reply]
No idea.  You've given me the absolute bare minimum to help you.  There is a whole setup document to follow and a code gen page you can use on shinylib.net

**Reply by manudev-1** [reply]
Sorry for this, i think i'll try to resolve the problem by myself, thank you anyway. have a nice day 😄

**Reply by aritchie** [reply]
just share a github link or something.  I'll look, but I need info
